react-basic-practice/src/components/CustomerTable/index.tsx

```diff
@@ -1,17 +1,58 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import {
   Th,
   Tr,
   Thead,
   Tbody,
   Td,
   Table as ChakraTable,
+  IconButton,
 } from "@chakra-ui/react";
-import { SortingIcon } from "@icons";
-import { ICustomerTableProps } from "@types";
+import { ICustomer, ICustomerTableProps, SortConfigType } from "@types";
 import TableRow from "../TableRow";
+import { SortingIcon } from "@icons";
+import { fetchUsers } from "@services";
+
+const CustomerTable: React.FC<ICustomerTableProps> = ({ columns, action }) => {
+  const [dataCustomer, setDataCustomer] = useState<ICustomer[]>([]);
+  const [sortConfig, setSortConfig] = useState<SortConfigType | undefined>();
+
+  /**
+   * Fetches user data from the API with the current sorting configuration and updates the state.
+   * @returns {Promise<void>} - A promise that resolves when the data has been fetched and state has been updated.
+   */
+  const fetchData = async () => {
+    try {
+      const result = await fetchUsers(sortConfig);
+      setDataCustomer(result);
+    } catch (error) {
+      console.error("Error fetching data:", error);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [sortConfig]);
+
+  /**
+   * Handles the sorting logic when a column header is clicked.
+   * @param {keyof ICustomer} key - The key of the column to sort by.
+   */
+  const handleSort = (key: keyof ICustomer) => {
+    setSortConfig((prevConfig) => {
+      if (prevConfig?.key === key) {
+        if (prevConfig.direction === "desc") {
+          return undefined;
```

Comment:
maybe `return` is enough, pls help to check.

Reply:
i done